src/main/scala/tech/cryptonomic/conseil/tezos/TezosPlatformDiscoveryOperations.scala

```diff
@@ -50,25 +43,25 @@ object TezosPlatformDiscoveryOperations {
 /** Class providing the implementation of the metadata calls with caching */
 class TezosPlatformDiscoveryOperations(
   metadataOperations: MetadataOperations,
-  attributesCache: MVar[IO, AttributesCache],
-  entitiesCache: MVar[IO, EntitiesCache],
-  attributeValuesCache: MVar[IO, AttributeValuesCache],
+  caching: MetadataCaching[IO],
   cacheOverrides: AttributeValuesCacheConfiguration,
-  cacheTTL: FiniteDuration)
+  cacheTTL: FiniteDuration,
+  networkName: String = "notUsed")
   (implicit executionContext: ExecutionContext) {
 
   import cats.effect._
   import cats.implicits._
 
+
   /** Method for initializing values of the cache */
   def init(): Future[Unit] = {
```

Comment:
I'm getting lost somewhat: who calls `init` and when?

Reply:
this is the first init which is initializing schema and sizes of tables, next is a manual init which initializes distinct counts(cardinality) of columns.